test/libsolidity/semanticTests/libraries/stub.sol

```diff
@@ -0,0 +1,13 @@
+library L {
+    function f(uint256 v) external returns (uint256) { return v*v; }
+}
+contract C {
+    function g(uint256 v) external returns (uint256) {
+        return L.f(v);
+    }
+}
+// ----
+// library: L
```

Comment:
What exactly is the syntax representing? That the lib used in `g` is `L`?

Reply:
This says "Deploy a copy of library L and link all contracts against it, i.e. set the address of the link references to L to the address of that deployed copy of L"...

Reply:
I thought about looking for all libraries in the source and deploying and resolving them automatically, but then we could for example not test that using only internal library functions doesn't require linking, etc...

Reply:
Ah, right. Can you add the same test but without this line?

Reply:
In fact I had that earlier, but ``soltest`` does have a check in place ensuring that the link references are empty when deploying the contract, so not having this line will fail an assertion....

Reply:
This one: https://github.com/ethereum/solidity/blob/50a015acf0c270d2cf9070d9e9d9178bd38a0dca/test/libsolidity/SolidityExecutionFramework.cpp#L86

Reply:
And at the moment we can't gracefully check for that error being thrown... would probably be quite hacky at the moment and involve catching that exception here and treating it specially... the problem is that ``SolidityExecutionFramework`` is still shared between the new extracted SemanticsTests and the old SolidityEndToEndTests.cpp ones... and for the latter I'd rather not remove this assertion... do you think it's worth spending some effort on this for adding that test?

Reply:
Ah ok, fine without. I guess we have tests for internal library calls without linking?

Reply:
Hm... we might actually not - no extracted ones for sure, but I don't see any non-extracted ones either... I'll add one here.

Reply:
Added.